src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -392,10 +392,12 @@ private void validatePartitionProperties(FailureCollector collector) {
     if (sourceTable == null) {
       return;
     }
-    TimePartitioning timePartitioning = ((StandardTableDefinition) (sourceTable).getDefinition())
-      .getTimePartitioning();
-    if (timePartitioning == null) {
-      return;
+    if (sourceTable.getDefinition() instanceof StandardTableDefinition) {
```

Comment:
can `sourceTable.getDefinition()` be instance of some other class? If so should we add a warn message saying that not supported or something similar?

Reply:
The other class is autogenerated by BQ, and is only the definition if table's data comes from an external (non-BQ) source, e.g. Google Sheets.

Reply:
I see.